weblogic-azure-aks/src/main/arm/scripts/applyGuaranteedQos.sh

```diff
@@ -0,0 +1,133 @@
+
+# get name of the running admin pod
+adminPodName=$(kubectl -n ${wlsDomainNS} get pod -l weblogic.serverName=admin-server -o json |
+    jq '.items[0] | .metadata.name' |
+    tr -d "\"")
+if [ -z "${adminPodName}" ]; then
+    echo_stderr "Fail to get admin server pod."
+    exit 1
+fi
+
+# run `source $ORACLE_HOME/wlserver/server/bin/setWLSEnv.sh > /dev/null 2>&1 && java weblogic.version` to get the version.
+# the command will print three lines, with WLS version in the first line.
+# use `grep "WebLogic Server" to get the first line.
+
+# $ source $ORACLE_HOME/wlserver/server/bin/setWLSEnv.sh > /dev/null 2>&1 && java weblogic.version
+# WebLogic Server 12.2.1.4.0 Thu Sep 12 04:04:29 GMT 2019 1974621
+# Use 'weblogic.version -verbose' to get subsystem information
+# Use 'weblogic.utils.Versions' to get version information for all modules
+rawOutput=$(kubectl exec -it ${adminPodName} -n ${wlsDomainNS} -c ${wlsContainerName} \
+    -- bash -c 'source $ORACLE_HOME/wlserver/server/bin/setWLSEnv.sh > /dev/null 2>&1 && java weblogic.version | grep "WebLogic Server"')
+
+# get version from string like "WebLogic Server 12.2.1.4.0 Thu Sep 12 04:04:29 GMT 2019 1974621"
+stringArray=($rawOutput)
+version=${stringArray[2]}
+echo_stdout "WebLogic Server version: ${version}"
+
+if [ "${version#*14.1.1.0}" != "$version" ]; then
+    timestampBeforePatchingDomain=$(date +%s)
+    echo  "timestampBeforePatchingDomain=${timestampBeforePatchingDomain}"
+    
+    # we assume the customer to create WebLogic Server using the offer or template,
+    # and specify the same resources requirement for admin server and managed server.
+    cpuRequest=$(kubectl get domain ${WLS_DOMAIN_UID} -n ${wlsDomainNS} -o json |
+        jq '. |.spec.serverPod.resources.requests.cpu' |
+        tr -d "\"")
+    echo_stdout "Previous CPU request: ${cpuRequest}"
+
+    memoryRequest=$(kubectl get domain ${WLS_DOMAIN_UID} -n ${wlsDomainNS} -o json |
+        jq '. | .spec.serverPod.resources.requests.memory' |
+        tr -d "\"")
+    echo_stdout "Previous memory request: ${memoryRequest}"
+
+    restartVersion=$(kubectl -n ${wlsDomainNS} get domain ${WLS_DOMAIN_UID} -o json |
+        jq '. | .spec.restartVersion' |
+        tr -d "\"")
+    restartVersion=$((restartVersion+1))
+
+    # check CPU units, set units with "m"
+    if [[ ${cpuRequest} =~ "m" ]]; then
+        cpu=$(echo $cpuRequest | sed 's/[^0-9]*//g')
+    else
+        cpu=$((cpuRequest * 1000))
+    fi
+    # make sure there is enough CPU limits to run the WebLogic Server
+    # if the cpu is less than 500m, set it 500m
+    # the domain configuration will be outputed after the offer deployment finishes.
+    if [ $cpu -lt 500 ]; then
+        cpu=500
+    fi
+
+    # create patch configuration with YAML file
+    # keep resources.limits the same with requests
+    cat <<EOF >patch-resource-limits.yaml
+spec:
+  serverPod:
+    resources: 
+      requests:
+        cpu: "${cpu}m"
+        memory: "${memoryRequest}"
```

Comment:
As far as I can tell, we are not modifying the `memoryRequest` from its previous value, retrieved on line 64, though we **are** modifying the `cpu` value.   Is that what we want?  I thought the essence of the workaround was to increase the **cpu** and **memory** requests and limits?

Reply:
Hello @edburns  per my observations, 1.5Gi is enough, so will not increase the memory in this workaround.